### DISCUSSION TITLE
Can UBL and mesh editing be considered in future releases?

### DISCUSSION THREAD

**Discussion by grrmisfit** [mriscoc/Ender3V2S1#30] [Ideas]
UBL being installed and manual tuning of the mesh. the firmware i ran on my v2 had both and is great for those slight adjustments that just happen. we all know the probe isnt perfect and sometimes it needs a little help, being able to go to each probe point and adjust each point up or down would help correct those

**Comment by mriscoc** [comment]
I could consider those additions, but with the current lack of flash space in the RCT6 chips I must prioritize most used features for the new capabilities.

**Comment by harriettubgirl** [comment]
I was also curious to know if you might add UBL. I'm just checking out your firmware, great work so far!! 

I had just gotten used to UBL so jumping into this has made me uncomfortable again :) Take care

edit: I had to stop using it for now, just don't have the time to mess with it at the moment. I really want to give it another shot sometime though, I was having issues figuring out how to save the mesh. I was trying to report the mesh for bedlevelvisualizer without having to re-run the 9x9 mesh. Wouldn't mind an easier way of handling meshes like a save/load mesh menu would be great. Thanks!

**Reply by mriscoc** [reply]
To retrieve the mesh for use with the Octoprint bed level visualizer: `M420 V`
To save the mesh from RAM to EEPROM: `Control/Store settings`
To get the mesh from EEPROM to RAM: `Control/Load settings`
To use of the mesh in RAM: `M420 S1 Z2`

**Reply by harriettubgirl** [reply]
Thanks for the reply! I'll have to try it out some more soon, take care
